llvm/lib/Target/AIE/AIEBasePipelinerLoopInfo.cpp

```diff
@@ -763,11 +768,23 @@ void ZeroOverheadLoop::adjustTripCount(int TripCountAdjust) {
 bool ZeroOverheadLoop::preferPostPipeliner(SMSchedule &SMS) {
   // Zero overhead loops are candidates for PostPipeliner, which does a better
   // job on multi-stage live-ranges without spilling or moving.
+
+  // PostPipeliner can do nothing without tripcount > 1
+  if (MinTripCount <= 1) {
+    return false;
+  }
```

Comment:
Hmm. It's a ZOL, Prepipeliner can't do anything either.

Reply:
Why not? We can add guards I think

Reply:
I feel that is hard to add guards in post, it will change the CFG and will require a new register for the constant...

Reply:
Without a loop counter GPR?